datafusion/src/logical_plan/expr.rs

```diff
@@ -1452,11 +1452,18 @@ impl fmt::Debug for Expr {
             }
             Expr::WindowFunction {
                 fun,
-                ref args,
+                args,
+                partition_by,
```

Comment:
It seems like the partition_by and order_by more logically belong on the `LogicalPlan::Window` node if they can be shared across `Expr::WindowFunction`

Reply:
that's a separate construct not yet available in the sql parser.

see https://www.postgresql.org/docs/current/tutorial-window.html

> When a query involves multiple window functions, it is possible to write out each one with a separate OVER clause, but this is duplicative and error-prone if the same windowing behavior is wanted for several functions. Instead, each windowing behavior can be named in a WINDOW clause and then referenced in OVER. For example:

```sql
SELECT sum(salary) OVER w, avg(salary) OVER w
  FROM empsalary
  WINDOW w AS (PARTITION BY depname ORDER BY salary DESC);
```

Reply:
> It seems like the partition_by and order_by more logically belong on the LogicalPlan::Window node if they can be shared across Expr::WindowFunction

for logically planned window function node it's not necessarily the case, because _sort order_ is defined by the concatenations of `partition_by` and then `order_by`, but consider:

```sql
select max(c1) over (partition by c2), max(c1) over (order by c2) from test;
```

both window functions will have the same sort key of `[c2 asc nulls first]` but they mean different things, and they might be logically planned together (meaning same pre-sorting happens) but not the same evaluation will happen

Reply:
I see -- this makes sense. Thank you for the clarification

Perhaps it is worth a comment (in a follow on PR) on the `LogicalPlan` window to this effect